src/visualizations/config/adapters/dhis_highcharts/getFilterText.js

```diff
@@ -1,68 +1,62 @@
-import { getOuFilterText } from '../../../../modules/ouFilterText'
+import { getOuFilterText } from '../../../../modules/getOuFilterText'
 
 export default function(filters, metaData) {
-    if (isArray(filters)) {
-        const titleParts = []
-        let i
-        let l
-
-        filters.forEach(filter => {
-            const items = dimensionGetItems(filter)
-            const hasOuLevel =
-                dimensionIs(filter, DIMENSION_ID_ORGUNIT) &&
-                items.some(item => ouIdHelper.hasLevelPrefix(item.id))
-            const hasOuGroup =
-                dimensionIs(filter, DIMENSION_ID_ORGUNIT) &&
-                items.some(item => ouIdHelper.hasGroupPrefix(item.id))
-
-            if (hasOuLevel || hasOuGroup) {
-                if (hasOuGroup) {
-                    titleParts.push(getOuFilterText(items, metaData, false))
-                }
-
-                if (hasOuLevel) {
-                    titleParts.push(getOuFilterText(items, metaData, true))
-                }
-            } else {
-                const filterItems = metaData.dimensions[filter.dimension]
-
-                if (isArray(filterItems)) {
-                    l = filterItems.length
-                    let id
-                    let sectionParts = []
-
-                    for (i = 0; i < l; i++) {
-                        id = filterItems[i]
-
-                        // if the value is present in items take the name to show from there
-                        if (metaData.items[id]) {
-                            sectionParts.push(metaData.items[id].name)
-                        }
-                        // otherwise use the values directly
-                        // this is a temporary fix to avoid app crashing when using filters with data items in EV
-                        else {
-                            sectionParts.push(
-                                metaData.items[filter.dimension].name +
-                                    ': ' +
-                                    filterItems.join(', ')
-                            )
+    if (!Array.isArray(filters) || !filters.length) {
+        return ''
+    }
 
-                            break
-                        }
+    const titleFragments = []
+    let i
+    let l
+
+    filters.forEach(filter => {
+        const items = dimensionGetItems(filter)
+
+        if (
+            dimensionIs(filter, DIMENSION_ID_ORGUNIT) &&
+            items.some(
+                ({ id }) =>
+                    ouIdHelper.hasGroupPrefix(id) ||
+                    ouIdHelper.hasLevelPrefix(id)
+            )
+        ) {
+            titleFragments.push(getOuFilterText(filter, metaData))
+        } else {
+            const filterItems = metaData.dimensions[filter.dimension]
+
+            if (Array.isArray(filterItems)) {
+                l = filterItems.length
+                let id
+                const sectionParts = []
+
+                for (i = 0; i < l; i++) {
```

Comment:
Maybe simplify this using `filterItems.forEach`?

Reply:
I left this code as is (it is the original implementation before I added anything), because it uses a `break`, which you cant do with foreach.

Reply:
Right!